### PR TITLE
PROGRAMMES-6761 Incorrect promotion size

### PR DESCRIPTION
**Pull request by Tasiobg**
- Updated the logics that builds the CSS for "content block promotions", this is now based on v2 code
- Updated the image presenter to use aspect ratio "auto" which mean the height will be propotional to the with we set, example /programmes/articles/5Mstn5KJJpgnH4Rd3SYHHSK/stories-and-features-about-the-archers

I could walk you thought the PR if you preffer an easy way to understand the changes

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -1,5 +1,5 @@
 includes:
-	- vendor/phpstan/phpstan-phpunit/extension.neon
+    - vendor/phpstan/phpstan-phpunit/extension.neon
```
Comment (Tasiobg) [inline phpstan.neon]: was a tab, now 4 spaces

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Ds2013/Presenters/Domain/ContentBlock/Promotions/PromotionsPresenter.php [hunk 2]
```diff
@@ -21,4 +21,20 @@ public function getPositionType(): string
 
         return 'subtle';
     }
+
```
Comment (Tasiobg) [inline src/Ds2013/Presenters/Domain/ContentBlock/Promotions/PromotionsPresenter.php]: This function mimic v2 behaviour

**Comment by Tasiobg** [comment]
Found missing code I lose when rebasing so I reopen the PR

File: phpstan.neon [hunk 3]
```diff
@@ -16,3 +16,4 @@ parameters:
         - '#Call to an undefined method Tests\\App\\Translate\\TranslatableTraitTest::localDateIntl().#'
         - '#Call to an undefined method BBC\\ProgrammesPagesService\\Domain\\Entity\\CoreEntity::getGenres().#'
         - '#Call to an undefined method BBC\\ProgrammesPagesService\\Domain\\Entity\\Programme::getReleaseDate()#'
+        - '#Call to an undefined method App\\ExternalApi\\Isite\\Domain\\ContentBlock\\AbstractContentBlock::getReleaseDate()#'
```
Comment (BjornTwachtmann) [inline phpstan.neon]: Does AbstractContentBlock have a getReleaseDate method? Don't quite understand this one

Comment (BjornTwachtmann) [reply phpstan.neon]: I get the following errors when running the tests:
```----- --------------------------------------------------------------------------------------------------------------
  Line   src/Ds2013/Presenters/Domain/ContentBlock/Promotions/PromotionsPresenter.php
 ------ --------------------------------------------------------------------------------------------------------------
  30     Call to an undefined method App\ExternalApi\Isite\Domain\ContentBlock\AbstractContentBlock::getPromotions().
 ------ --------------------------------------------------------------------------------------------------------------

 --------------------------------------------------------------------------------------------
  Error
 --------------------------------------------------------------------------------------------
  Ignored error pattern #Call to an undefined method
  App\\ExternalApi\\Isite\\Domain\\ContentBlock\\AbstractContentBlock::getReleaseDate()# was
  not matched in reported errors.
 --------------------------------------------------------------------------------------------```

Comment (Tasiobg) [reply phpstan.neon]: I believe that is an undo I commited by mistake

**Pull request merged**